rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java

```diff
@@ -275,13 +276,54 @@ private void testChangePowpeg(
         /*
           Activation phase
          */
-
         // Move the required blocks ahead for the new powpeg to become active
         // (overriding block number to ensure we don't move beyond the activation phase)
-        blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge();
+        blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge(activations);
         Block activationBlock = mock(Block.class);
         doReturn(blockNumber).when(activationBlock).getNumber();
 
+        // assuming fed activation age after rskip383 is greater than legacy fed activation age, we can check that new fed
+        // should not be active at the legacy activation age when RSKIP383 is active
+
+        if (activations.isActive(ConsensusRule.RSKIP383)){
+            ActivationConfig.ForBlock activationsBeforeRSKIP383 = mock(ActivationConfig.ForBlock.class);
+            when(activationsBeforeRSKIP383.isActive(ConsensusRule.RSKIP383)).thenReturn(false);
+
+            long beforeRskip383blockNumber = initialBlock.getNumber() + bridgeConstants.getFederationActivationAge(activationsBeforeRSKIP383);
+            Assertions.assertTrue(blockNumber > beforeRskip383blockNumber);
+
+            Block beforeRskip383ActivationBlock = mock(Block.class);
+            doReturn(beforeRskip383blockNumber).when(beforeRskip383ActivationBlock).getNumber();
+
+            bridgeSupport = new BridgeSupportBuilder()
+                .withProvider(bridgeStorageProvider)
+                .withRepository(repository)
+                .withEventLogger(bridgeEventLogger)
+                .withExecutionBlock(beforeRskip383ActivationBlock)
+                .withActivations(activations)
+                .withBridgeConstants(bridgeConstants)
+                .withBtcBlockStoreFactory(btcBlockStoreFactory)
+                .withPeginInstructionsProvider(new PeginInstructionsProvider())
+                .build();
+
+            // New active powpeg and retiring powpeg
+            assertNotEquals(newPowPegAddress, bridgeSupport.getFederationAddress());
+            assertEquals(oldPowPegAddress, bridgeSupport.getFederationAddress());
+            assertNotEquals(oldPowPegAddress, bridgeSupport.getRetiringFederationAddress());
```

Comment:
What's the idea here? Proving that the new fed is not yet active?

Reply:
Refactored